### PR TITLE
Implementation of displaying multiple stats at once

### PR DESCRIPTION
**Pull request by riuson**
For issue #5.

Screenshot 2021-09-04 223743

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,16 +58,25 @@
...
-          ],
-          "description": "Choose the statistic item that is shown in the status bar."
+        "markyMarkdown.statsStatusBarShowReadingTime": {
+          "type": "boolean",
+          "default": true,
+          "description": "Show Reading Time in the status bar."
+        },
+        "markyMarkdown.statsStatusBarShowWords": {
+          "type": "boolean",
+          "default": true,
```
Comment (robole) [inline package.json]: change to false

File: package.json [hunk 2]
```diff
@@ -58,16 +58,25 @@
...
+          "description": "Show Words counter in the status bar."
+        },
+        "markyMarkdown.statsStatusBarShowLines": {
+          "type": "boolean",
+          "default": true,
```
Comment (robole) [inline package.json]: change to false

File: package.json [hunk 3]
```diff
@@ -58,16 +58,25 @@
...
+          "description": "Show Lines counter in the status bar."
+        },
+        "markyMarkdown.statsStatusBarShowCharacters": {
+          "type": "boolean",
+          "default": true,
```
Comment (robole) [inline package.json]: change to false

File: src/statisticPicker.js [hunk 4]
```diff
@@ -3,15 +3,28 @@
 const vscode = require("vscode");
 const activeDoc = require("./activeDocument");
 
+const panels = {
```
Comment (robole) [inline src/statisticPicker.js]: _labels_ is a more appropriate name

File: src/statisticPicker.js [hunk 5]
```diff
@@ -21,24 +34,32 @@ class StatisticPicker {
...
-    this.statusBarItem.text = this.quickPickItems[this.selection].label;
+    var filteredItems = this.quickPickItems.filter(
+      (x) => this.selectedItems.indexOf(x.name) >= 0
+    );
+
+    filteredItems.forEach(function (item, index) {
+      item.picked = true;
+    });
+
+    var label = filteredItems.map((x) => x.label).join(" | ");
```
Comment (robole) [inline src/statisticPicker.js]: I would prefer to have 2 spaces between items, rather than using the spaces with a pipe character

**Review by robole — COMMENTED** [review]
hi @riuson,

thanks for the PR. 

i had a quick look and it looks quite good. i have left a few comments to tidy a few bits up. its important to minimize the impact on existing users, so changing the default option may annoy users, only "reading time" should be selected by default to be consistent with the current setting.

can you finish the related documentation too when you get a chance? The README  has a _Settings_ section that needs to reflect the new variables, and the changelog requires an entry. If you are busy I understand, just let me know. I don't have much time to work on side projects like this, so I  must be selective in what I do.

I can update the screenshot in the README and give it a test before I publish it to the marketplace. It may take me a while to come back to it. You are of course welcome to publish your own version if you prefer. It is up to you.

All the best

Rob

File: package.json (hunk 1 above)
Comment (riuson) [reply package.json]: Fixed

File: package.json (hunk 2 above)
Comment (riuson) [reply package.json]: Fixed

File: package.json (hunk 3 above)
Comment (riuson) [reply package.json]: Fixed

File: src/statisticPicker.js (hunk 4 above)
Comment (riuson) [reply src/statisticPicker.js]: Fixed

File: src/statisticPicker.js (hunk 5 above)
Comment (riuson) [reply src/statisticPicker.js]: Added configurable separator with default 2 spaces.

**Comment by riuson** [comment]
Readme & screenshot are updated.
The version number and the line in the changelog are at your discretion.
example

**Pull request merged**